internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/public.go

```diff
@@ -457,23 +464,30 @@ func (h *Handler) updatePublicShare(w http.ResponseWriter, r *http.Request, shar
 	}
 
 	// Password
-	newPassword, ok := r.Form["password"]
+	_, ok = r.Form["password"]
+	newPassword := strings.TrimSpace(r.FormValue("password"))
```

Comment:
Why is it necessary to trim spaces? I wouldn't do any magic on the password the user provided

Reply:
fixed.